beacon-chain/sync/service.go

```diff
@@ -100,10 +103,33 @@ type Service struct {
 	blockNotifier        blockfeed.Notifier
 	blocksRateLimiter    *leakybucket.Collector
 	attestationNotifier  operation.Notifier
+	seenBlockLock        sync.RWMutex
+	seenBlockCache       *ristretto.Cache
+	seenAttestationLock  sync.RWMutex
+	seenAttestationCache *ristretto.Cache
 }
 
 // Start the regular sync service.
 func (r *Service) Start() {
+	bCache, err := ristretto.NewCache(&ristretto.Config{
```

Comment:
I think another cache might have to be used for this, in ristretto `Sets` aren't guaranteed and for this usecase, it seems like we would want setting a value to be included in the cache

Reply:
Good point, switched to LRU. Please take a look